### PR TITLE
enable multi-network PlacementConstrains

### PR DESCRIPTION
**Pull request by yastij**
Signed-off-by: Yassine TIJANI <ytijani@vmware.com>

**What this PR does / why we need it**: This PR adds support for zone using multiple networks. The list of network are applied in order to list of `DeviceSpecs` of the machine

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

/assign @gab-satchi @srm09 

**Release note**:

```release-note

```

### CODE REVIEW

**Review by yastij — COMMENTED** [review]
/approve

File: api/v1alpha4/vspheredeploymentzone_types.go [hunk 1]
```diff
@@ -50,15 +51,29 @@ type PlacementConstraint struct {
 	// +optional
 	Datastore string `json:"datastore,omitempty"`
 
-	// NetworkName is the network name for this machine's VM.
-	NetworkName string `json:"NetworkName"`
+	// Network represents the networking for this depoyment zone
+	// +optional
+	Network []Network `json:"Network,omitempty"`
```
Comment (srm09) [inline api/v1alpha4/vspheredeploymentzone_types.go:56]: Same as above.

**Review by srm09 — COMMENTED** [review]
I have a question about the length of the `Network` array. Otherwise lgtm.

**Pull request merged**